### PR TITLE
🐛(poll) fix bug select

### PR DESCRIPTION
**Pull request by Kitero**
## Purpose

Fix of a bug on the select to add a new field which didn't set the right type and widget name.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unrealeased]
 
+### Changed
```
Comment (jbpenrath) [inline CHANGELOG.md:11]:
```suggestion
### Fixed
```

File: examples/poll/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (jbpenrath) [inline examples/poll/jest.config.js:1]: Can you please move all jest configuration stuff in a dedicated commit.

File: examples/poll/jest/resolver.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      // implementation is available for the browser+ESM version of uuid to use (eg, via
+      // https://github.com/jsdom/jsdom/pull/3352 or a similar polyfill), this can go away.
+      if (['nanoid', 'uuid'].includes(pkg.name)) {
+        delete pkg['exports'];
+        delete pkg['module'];
+      }
+      return pkg;
+    },
+  });
+};
```
Comment (jbpenrath) [inline examples/poll/jest/resolver.js:26]: You did not use this file through jest.config.

File: examples/poll/src/components/Modal/AddQuestionModal.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import { VernaForm, VernaProvider } from '@openfun/verna';
+import * as factories from '@openfun/verna/src/tests/mocks/factories';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { Suspense } from 'react';
+import AddQuestionModal from ':/components/Modal/AddQuestionModal';
+
+describe('feature to add widgets using a modal and a select', () => {
```
Comment (jbpenrath) [inline examples/poll/src/components/Modal/AddQuestionModal.spec.tsx:8]: Please use module name as test suite name.
```suggestion
describe('AddQuestionModal', () => {
```

File: examples/poll/src/components/Modal/AddQuestionModal.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      await userEvent.click(screen.getByRole('button', { name: 'save' }));
+    });
+
+    // Reopen parameters
+    await act(async () => {
+      await userEvent.click(screen.getAllByRole('button', { name: 'Parameters' })[3]);
+    });
+
+    expect(screen.getAllByRole('textbox', {})[1]).toHaveValue('test');
+  });
```
Comment (jbpenrath) [inline examples/poll/src/components/Modal/AddQuestionModal.spec.tsx:76]: I think a better test would be to test if schema has been updated properly once the user clicks to add the widget.

File: examples/poll/src/components/Modal/AddQuestionModal.tsx [hunk 6]
```diff
@@ -43,10 +52,47 @@ export default function AddQuestionModal({
...
+        widgetName: 'checkboxesWidget',
+      },
+      {
+        text: intl.formatMessage(SelectWidgetMessages.SelectWidget),
+        type: 'string',
+        widgetName: 'selectWidget',
+      },
+    ],
+    [intl, locale],
+  );
```
[the remark this answers is not in the document]
Comment (Kitero) [reply examples/poll/src/components/Modal/AddQuestionModal.tsx:94]: I know but in the toolbar the translations are displayed directly in the jsx, so we won't have it here, at best i can have the type and widgetName but we would need to order informations to add the right translations

[the remark this answers is not in the document]
Comment (jbpenrath) [reply examples/poll/src/components/Modal/AddQuestionModal.tsx:94]: So are you saying we are able to remove all widgetToolbarItems then just iterate over as follow ? 😛 

```jsx
<VernaToolbar>
  {widgets.map(widget => <WidgetWrapper key={widget.widgetName} {...widget} />}
</VernaToolbar>
```

File: examples/poll/jest/resolver.js (hunk 3 above)
Comment (Kitero) [reply examples/poll/jest/resolver.js:26]: Indeed i don't need it

File: examples/poll/src/components/Modal/AddQuestionModal.spec.tsx (hunk 5 above)
Comment (Kitero) [reply examples/poll/src/components/Modal/AddQuestionModal.spec.tsx:76]: Indeed it's more reliable i'll add one

File: examples/poll/jest.config.js (hunk 2 above)
Comment (jbpenrath) [inline examples/poll/jest.config.js:1]: In your commit message.

You do not add resolver anymore.

File: examples/poll/src/components/Modal/AddQuestionModal.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (Kitero) [reply examples/poll/src/components/Modal/AddQuestionModal.tsx:94]: image

File: examples/poll/src/components/FormWrapper/FormWrapper.tsx [hunk 7]
```diff
@@ -78,6 +74,7 @@ export default function FormWrapper({ setIsEditor }: FormWrapperProps) {
     () => isEditor && pollSelected !== null,
     [isEditor, pollSelected],
   );
+  const widgets = useMemo(() => getWidgets(intl), [locale, intl]);
```
Comment (jbpenrath) [inline examples/poll/src/components/FormWrapper/FormWrapper.tsx:77]: locale is not dependency require by getWidgets.

Comment (Kitero) [reply examples/poll/src/components/FormWrapper/FormWrapper.tsx:77]: No but i want to reload the memo if we change the language, otherwise it may keep the translations of the previous locale, i'm really doubting the intl dep is enougth..

Comment (Kitero) [reply examples/poll/src/components/FormWrapper/FormWrapper.tsx:77]: Ok it does works :+1:

Comment (jbpenrath) [reply examples/poll/src/components/FormWrapper/FormWrapper.tsx]: Yep you can retrieve the active locale through `intl.locale`

File: examples/poll/src/components/Modal/AddQuestionModal.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+    // Check that the newly created widget is well initialized
+    const newWidget = useVernaValue?.uiSchema?.['testSection']['ui:order'].filter(
+      (widget: string) => widget !== 'field1',
+    )[0];
+    expect(newWidget).toBeDefined();
+    expect(
+      useVernaValue?.formSchema?.['properties']?.['testSection']?.['properties']?.[newWidget]?.type,
+    ).toBe('string');
+  });
```
Comment (jbpenrath) [inline examples/poll/src/components/Modal/AddQuestionModal.spec.tsx:118]: I'll complete the test as follow (check that the right ui widget has been added to uiSchema)
```suggestion
    // Check that the newly created widget is well initialized
    const { formSchema, uiSchema } = useVernaValue!;
    const newWidgetId = uiSchema?.['testSection']['ui:order'].find(
      (widget: string) => widget !== 'field1',
    );
    expect(newWidgetId).toBeDefined();
    expect(uiSchema?.['testSection'][newWidgetId]).toEqual({ 'ui:widget': 'selectWidget' });

    expect(formSchema?.['properties']?.['testSection']?.['properties']?.[newWidgetId]?.type).toBe(
      'string',
    );
```

**Review by jbpenrath — APPROVED** [review]
